src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -76,7 +75,7 @@ public NatsConnection()
     public NatsConnection(NatsOpts opts)
     {
         _logger = opts.LoggerFactory.CreateLogger<NatsConnection>();
-        Opts = opts;
+        Opts = ReadUserInfoFromConnectionString(opts);
```

Comment:
I'm not a fan of this form. You are both modifying the value and returning/assigning it. I'd prefer a "procedure" versus a function. Maybe a private setter?

Reply:
since we're not using a builder it'd be quite a big change trying to create `NatsOpts` once. so the pattern here is using the immutable `opts` object and a pure function `ReadUserInfoFromConnectionString()` to return a new `opts` object with the desired properties set to new values.